internal/driver/onvifclient.go

```diff
@@ -418,25 +417,12 @@ func (onvifClient *OnvifClient) callSubscribeCameraEventFunction(resourceName, s
 
 // callGetSnapshotFunction returns a snapshot from the camera as a slice of bytes
 // The implementation can refer to https://github.com/edgexfoundry/device-camera-go/blob/5c4f34d1d59b8e25e1a6316661d463e2495d45fe/internal/driver/onvifclient.go#L119
-func (onvifClient *OnvifClient) callGetSnapshotFunction() ([]byte, errors.EdgeX) {
-	// Get the token from the profile
-	respContent, edgexErr := onvifClient.callOnvifFunction(onvif.MediaWebService, onvif.GetProfiles, nil)
-	if edgexErr != nil {
-		return nil, errors.NewCommonEdgeXWrapper(edgexErr)
-	}
-	profilesResp, ok := respContent.(*media.GetProfilesResponse)
-	if !ok {
-		return nil, errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("invalid GetProfilesResponse of type %T for the camera %s", respContent, onvifClient.DeviceName), nil)
-	}
-	if len(profilesResp.Profiles) == 0 {
-		return nil, errors.NewCommonEdgeX(errors.KindServerError, "no onvif profiles found", nil)
-	}
-	requestData, edgexErr := snapshotUriRequestData(profilesResp.Profiles[0].Token)
-	if edgexErr != nil {
-		return nil, errors.NewCommonEdgeXWrapper(edgexErr)
+func (onvifClient *OnvifClient) callGetSnapshotFunction(profileToken []byte) ([]byte, errors.EdgeX) {
```

Comment:
`profileToken` might be misleading here, as the byte array is a json object of: 

```json
{"ProfileToken":"Profile_1"}
```

So just calling it data or something like that is fine I would think, and if we wanted, we could add a comment that it should contain the profile token

Reply:
But that json object contains only profileToken. I can change it tough.

Reply:
what i mean is that i correlate a parameter called "profileToken" to be a strong value of the token, and not a json object containing the token as one of the fields.